w3af/plugins/grep/analyze_cookies.py

```diff
@@ -187,6 +199,27 @@ def _analyze_cookie_security(self, request, response, cookie_obj,
         self._http_only(request, response, cookie_obj,
                         cookie_header_value, fingerprinted)
 
+    def _update_cookie_vulnerability(self, response, vulnerability):
+        # thread lock to escape race condition on db access
+        with self._plugin_lock:
+            vulns = kb.kb.get(self, 'security')
+        
+            if len(vulns) == 0:
+                kb.kb.append(self, 'security', vulnerability)
+            elif len(vulns) >= 1:
+                old_vuln = vulns[0]
+                updated_vuln = copy.deepcopy(old_vuln)
+                if 'urls' not in updated_vuln.keys():
+                    updated_vuln['urls'] = [old_vuln.get_url()]
+                updated_vuln['urls'].append(response.get_url())
+                
+                kb.kb.update(old_vuln, updated_vuln)
+            else:
+                # error
+                raise ex.DBException('An error has occured retrieving' \
```

Comment:
I believe there is no way to reach this line of code. Remove all the "else" section of this if statement after confirming.

Also remove the `ex` import above.